frontend/lib/resolution/default-functions.cpp

```diff
@@ -195,12 +195,13 @@ generateInitParts(Context* context,
   if (compType->isRecordType() || compType->isUnionType()) {
     qtReceiver = QualifiedType(QualifiedType::REF, compType);
 
-  // If the receiver is a basic class C, use 'const in x: borrowed C'.
+  // If the receiver is a basic class C, use 'const in x: borrowed C?'.
   } else if (auto basic = compType->toBasicClassType()) {
     const Type* manager = nullptr;
-    auto nonNilBorrowed  = ClassTypeDecorator::BORROWED_NONNIL;
-    auto decor = ClassTypeDecorator(nonNilBorrowed);
-    auto receiverType = ClassType::get(context, basic, manager, decor);
+    auto borrowedNilableDecor =
+        ClassTypeDecorator(ClassTypeDecorator::BORROWED_NILABLE);
+    auto receiverType =
+        ClassType::get(context, basic, manager, borrowedNilableDecor);
```

Comment:
I don't think we allow calling methods/initializers on a nilable class, so I don't think this is accurate. Is this part of the change to support initializing nilable from non-nilable?

Reply:
> I don't think we allow calling methods/initializers on a nilable class

I didn't know this, so the change could definitely be wrong.

> Is this part of the change to support initializing nilable from non-nilable?

Sort of, it actually is apparently required to support _deinitializing_ a nilable, which is the only thing that breaks if I switch it back. I figure deinit of nilable classes is something we need to do and I'm not sure how else to enable it; maybe it needs to be more of a special case?

Reply:
In the generated code we should eventually do a runtime check to see whether the variable being initialized is actually ``nil``, and will only call ``deinit`` if it isn't ``nil``. So for the purposes of resolving the class's ``deinit`` I think we can rely on a nilable->non-nilable conversion before invoking the method. Updating the invocation of ``deinit`` to use a non-nilable class might suffice to fix this problem.

Reply:
Makes sense, done and it seems to work. Thanks